src/main/java/tech/stackable/hadoop/StackableAccessControlEnforcer.java

```diff
@@ -19,153 +25,202 @@
-
-        // Guaranteed to be only called once (Effective Java: Item 3)
-        Configuration configuration = HadoopConfigSingleton.INSTANCE.getConfiguration();
-
-        String opaPolicyUrl = configuration.get(OPA_POLICY_URL_PROP);
-        if (opaPolicyUrl == null) {
-            throw new OpaException.UriMissing(OPA_POLICY_URL_PROP);
-        }
-
-        try {
-            this.opaUri = URI.create(opaPolicyUrl);
-        } catch (Exception e) {
-            throw new OpaException.UriInvalid(opaUri, e);
-        }
-
-        this.json = new ObjectMapper()
-                // OPA server can send other fields, such as `decision_id`` when enabling decision logs
-                // We could add all the fields we *currently* know, but it's more future-proof to ignore any unknown fields
-                .configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false)
-                // Previously we were getting
-                // Caused by: com.fasterxml.jackson.databind.exc.InvalidDefinitionException: No serializer found for class org.apache.hadoop.hdfs.util.EnumCounters and no properties discovered to create BeanSerializer (to avoid exception, disable SerializationFeature.FAIL_ON_EMPTY_BEANS) (through reference chain: tech.stackable.HdfsOpaAccessControlEnforcer$ContextWrapper["inodeAttrs"]->org.apache.hadoop.hdfs.server.namenode.INodeDirectory[0]->org.apache.hadoop.hdfs.server.namenode.INodeDirectory["features"]->org.apache.hadoop.hdfs.server.namenode.DirectoryWithQuotaFeature[0]->org.apache.hadoop.hdfs.server.namenode.DirectoryWithQuotaFeature["spaceConsumed"]->org.apache.hadoop.hdfs.server.namenode.QuotaCounts["typeSpaces"])
-                .configure(SerializationFeature.FAIL_ON_EMPTY_BEANS, false)
-                // Only include the needed fields. HDFS has many classes with even more circular reference to remove
-                .setVisibility(PropertyAccessor.ALL, JsonAutoDetect.Visibility.NONE)
-                .setVisibility(PropertyAccessor.FIELD, JsonAutoDetect.Visibility.PUBLIC_ONLY)
-                .setVisibility(PropertyAccessor.GETTER, JsonAutoDetect.Visibility.PUBLIC_ONLY)
-                // We need to remove some circular pointers (e.g. root -> children[0] -> parent -> root)
-                // Otherwise we get com.fasterxml.jackson.databind.JsonMappingException: Infinite recursion (StackOverflowError)
-                .addMixIn(DatanodeDescriptor.class, DatanodeDescriptorMixin.class);
-
-        LOG.debug("Started HdfsOpaAccessControlEnforcer");
+public class StackableAccessControlEnforcer
+    implements INodeAttributeProvider.AccessControlEnforcer {
+
+  private static final Logger LOG = LoggerFactory.getLogger(StackableAccessControlEnforcer.class);
+
+  public static final String OPA_POLICY_URL_PROP = "hadoop.security.authorization.opa.policy.url";
+
+  private static final HttpClient HTTP_CLIENT =
+      HttpClient.newBuilder().connectTimeout(Duration.ofSeconds(30)).build();
```

Comment:
Was there any need for a custom timeout or is this "just" considered best practice?

Reply:
"just" best practice